File: mathematics/sq_percision.cpp
/* author : @akash */

#include<bits/stdc++.h>
using namespace std;

#define ll long long int
#define pb push_back
#define mod 1000000007
#define ld long double


void solve()
{
   ll n;
   ll p;
   cin>>n;
   cin>>p;
   float ans=sqrt(n);
   cout<<ans<<"\n";
   if(p==1)
   {
      printf(".%1f",ans);
   }
   else if(p==2)
   {
   	printf(".%2f",ans);
   }
   else if(p==3)
   {
   	printf("%.3f",ans);
   }
   else
   {
   	printf("%.4f",ans);
   }
}

int main()
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);
    solve();
    cout<<"\n";	    
	return 0;
}

File: mathematics/sq_percision.py
#author : @akash

n=int(input())
p=int(input())
import math
ans=math.sqrt(n)
ans=round(ans,p+1)
if p==1:
   ans=ans*100
elif p==2:
   ans*=1000
elif p==3:
   ans*=10000
else:
   ans*=100000
ans=ans//10
if p==1:
  ans=ans/10
elif p==2:
   ans=ans/100
elif p==3:
   ans=ans/1000
else:
   ans=ans/10000
print(ans)
